MonMaraicher/Features/FarmerDetail/FarmerDetailsView.swift

```diff
@@ -160,39 +119,91 @@ private extension FarmerDetailsView {
                 }
                 Spacer()
                 phoneButton
-                    .frame(width: 80, height: 35)
-                    .background(.ultraThinMaterial)
-                    .clipShape(.capsule)
-                    .padding()
-                    .shadow(radius: 4)
             }
             Divider()
-            VStack(alignment: .leading) {
-                Text("Email")
-                    .foregroundStyle(.secondary)
-                    .font(.subheadline)
-                Link(viewModel.email!, destination: URL(string: viewModel.email!)!)
+            HStack {
+                VStack(alignment: .leading) {
+                    Text("Email")
+                        .foregroundStyle(.secondary)
+                        .font(.subheadline)
+                    Text(viewModel.email ?? "Aucune adresse email")
+                        .font(.callout)
+                }
+                Spacer()
+                mailButton
             }
             Divider()
-            VStack(alignment: .leading) {
+            VStack(alignment: .leading, spacing: 5) {
+                Text("Site Web")
+                    .foregroundStyle(.secondary)
+                    .font(.subheadline)
                 if !viewModel.websites.isEmpty {
-                    Text("Site Web")
-                        .foregroundStyle(.secondary)
-                        .font(.subheadline)
                     ForEach(viewModel.websites, id: \.id) { website in
-                        Link(website.websiteType.name, destination: URL(string: website.url)!)
+                        Text(.init("[\(website.websiteType.name)](\(website.url))"))
```

Comment:
Why not having that string building inside your viewModel instead? I could be unit tested then. In your view it will never be tested

Reply:
Good idea!